### PR TITLE
Allow passing dims to Potential and Deterministic

### LINKED ISSUE #6544: Allow passing dims to `Potential`
Not important for InferenceData, since potentials don't produce any values that remain in the trace (unless a user wraps them in a Deterministic, which already accept dims), but it would be neat for `graphviz`.

It's a simple as forwarding dims here: https://github.com/pymc-devs/pymc/blob/562e51edf741e704ccdf755718a2cff6da217efb/pymc/model.py#L2052

### PR DESCRIPTION
**Pull request by Raj-Parekh24**
**What is this PR about?**
Closes #6544.

**Checklist**
+ [x] Explain important implementation details 👆
+ [x] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [x] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [x] Are the changes covered by tests and docstrings?
+ [x] Fill out the short summary sections 👇

## Major / Breaking Changes
- ...

## New features
- Add dims parameter in Potential function defined in the model.py file.

## Bugfixes
- ...

## Documentation
- ...

## Maintenance
- ...

### CODE REVIEW

**Comment by Raj-Parekh24** [comment]
Closes #6544

**Comment by ricardoV94** [comment]
Thanks @Raj-Parekh24

Can you confirm dims are saved in "model.named_vars_to_dims”? There must be a test that already checks this for deterministics, so you can put it nearby.

**Comment by Raj-Parekh24** [comment]
Sure @ricardoV94 ,

I will look into it and add appropriate test case.

**Comment by Raj-Parekh24** [comment]
Hi @ricardoV94,

Added test case "test_potential_with_dims" to check the dims passing to Potential please review it.

File: tests/test_model.py [hunk 1]
```diff
@@ -1062,6 +1062,16 @@ def test_deterministic():
     assert model["y"] == y
 
 
+def test_potential_with_dims():
+    """
+    Test to check the passing of dims to the potential
+    """
+    with pm.Model(coords={"observed": range(10)}) as model:
+        x = pm.Normal("x", 0, 1)
+        y = pm.Deterministic("y", x**2, dims=("observed",))
```
Comment (michaelosthege) [inline tests/test_model.py:1071]: This test does not use `Potential`

Comment (Raj-Parekh24) [reply tests/test_model.py:1071]: Thank you for the review @michaelosthege 

Sorry for this,

I found that dims test case was not available for deterministic as well so added that test case along with potential.

I might forget to update the test case for potential.

I have updated the code.

Please review it.

**Review by ricardoV94 — APPROVED** [review]
Thanks @Raj-Parekh24

**Review by michaelosthege — APPROVED** [review]

**Pull request merged**